_rules/image-has-name-23a2a8.md

```diff
@@ -24,7 +24,7 @@ The rule applies to HTML `img` elements or any HTML element with the [semantic r
 
 ## Expectation
 
-Each target element has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace), or is marked as [decorative](#decorative).
+Each target element has an [accessible name](#accessible-name) that consists of characters other than [whitespace](#whitespace), or is marked as [decorative](#decorative).
```

Comment:
That seems to change the meaning. `"   a   "` is not only whitespaces but I'm not sure we can say that it "consists of characters other than whitespace" (since there are also whitespaces in it).
And `"A picture of a dolphin"` does contain whitespaces but is definitely a perfectly valid name for an image.

Something in the line of "contains at least one character which is not a whitespace" would be better (and keep the same meaning as the original one), but that still contains a negative, only pushed a bit further…

Reply:
I agree that "consists of characters other than whitespace" is ambiguous. It could mean "this string only consists of characters other than whitespace" or it could mean "this string consists of at least 1 character other than whitespace".

This ambiguity should be resolved.

Reply:
I agree that in those cases, trying to reverse the negative in a positive way do not help the reading, more, it can make it harder to understand and leads to mistakes.